bodhi/server/consumers/masher.py

```diff
@@ -623,56 +687,106 @@ def remove_pending_tags(self):
+        template = env.get_template(config_template)
+
+        pungi_conf_dir = tempfile.mkdtemp(prefix='bodhi-pungi-%s-' % self.id)
+
+        with open(os.path.join(pungi_conf_dir, 'pungi.conf'), 'w') as conffile:
+            conffile.write(template.render())
+
+        self.copy_additional_pungi_files(pungi_conf_dir, env)
+
+        return pungi_conf_dir
 
     def mash(self):
-        if self.path in self.state['completed_repos']:
+        if self.path and self.path in self.state['completed_repos']:
             self.log.info('Skipping completed repo: %s', self.path)
             return
 
-        comps = os.path.join(self.my_comps_dir, 'comps-%s.xml' %
-                             self.release.branch)
-        previous = os.path.join(config.get('mash_stage_dir'), self.id)
 
-        mash_thread = MashThread(self.id, self.path, comps, previous, self.log)
-        mash_thread.start()
-        return mash_thread
+        previous = self.get_previous_compose()
+        config_dir = self.create_pungi_config()
+        config_file = os.path.join(config_dir, 'pungi.conf')
+        pungi_cmd = [config.get('pungiconf.cmd'),
+                     '--config', config_file,
+                     '--no-label',
+                     '--target-dir', self.mash_dir]
+        if previous:
+            pungi_cmd += ['--old-composes', previous]
+
+        self.log.info('Running the pungi command: %s', pungi_cmd)
+        mash_process = subprocess.Popen(pungi_cmd,
+                                        # Nope. No shell for you
+                                        shell=False,
+                                        # Should be useless, but just to set something predictable
+                                        cwd=self.mash_dir,
+                                        # Pungi will logs its stdout into pungi.global.log
+                                        stdout=DEVNULL,
+                                        # Stderr should also go to pungi.global.log if it starts
+                                        stderr=subprocess.PIPE,
+                                        # We will never have additional input
+                                        stdin=DEVNULL)
+        self.log.info('Pungi running as PID: %s', mash_process.pid)
+        # Since the mash process takes a long time, we can safely just wait 3 seconds to abort the
+        # entire mash early if Pungi fails to start up correctly.
+        time.sleep(3)
+        if mash_process.poll() not in [0, None]:
+            self.log.error('Pungi process terminated with error within 3 seconds! Abandoning!')
+            _, err = mash_process.communicate()
+            self.log.error('Stderr: %s', err)
+            raise Exception('Pungi returned error, aborting!')
+
+        return mash_process
+
+    def get_previous_compose(self):
+        """Function to return the previous composes' output."""
+        return None
+        return os.path.join(config.get('mash_stage_dir'), self.id)
```

Comment:
Is the intention to kill this method, or was this ```return None``` just here for debugging?

Reply:
This was for debugging.